CMakeLists.txt

```diff
@@ -141,6 +142,19 @@ elseif (POSIX)
 	test_requires_clock_gettime(ENABLE_MONOTONIC_CLOCK_DEFAULT MONOTONIC_CLOCK_LINKLIB)
 endif()
 
+# If current platform is android both ENABLE_MONOTONIC_CLOCK_DEFAULT is ON so check
+# todo: This is not a very good approach, but in the current project,
+# todo: the platform is not defined independently, and there is a mix of platform and ENABLE_MONOTONIC_CLOCK in src/core/sync.h,
+# todo: which should not be used together. Therefore, for now, we have to write it this way.
+if (ANDROID AND ENABLE_MONOTONIC_CLOCK_DEFAULT)
```

Comment:
I don't think you need to make any special check if the problem is on Android. Sure, you need this above comment, but you can as well check BOTH things - availability of monotonic clock and availability of `pthread_condattr_setclock` in one step (the above if-branch)

Checks should be done elsewhere (after the option series, where you'd expect to define the ENABLE_MONOTONIC_CLOCK option). What you need to check is:
* if ENABLE_STDCXX_SYNC is OFF, you don't check anything
* Otherwise, if ENABLE_MONOTONIC_CLOCK is ON, you check if ENABLE_MONOTONIC_CLOCK_DEFAULT and HAVE_PTHREAD_CONDATTR_SETCLOCK are both ON. If not, you issue a FATAL_ERROR.

Reply:
hmmm, 

"if ENABLE_STDCXX_SYNC is OFF, you don't check anything"
Is there a mistake here? Shouldn’t it be that when ENABLE_STDCXX_SYNC is ON, I don’t need to do any checks because when ENABLE_STDCXX_SYNC is ON, it will use C++11.

Only when ENABLE_STDCXX_SYNC is OFF, and the current platform is Android (because of the comment above), and ENABLE_MONOTONIC_CLOCK is ON, will I check ENABLE_MONOTONIC_CLOCK_DEFAULT and HAVE_PTHREAD_CONDATTR_SETCLOCK.

If my understanding is correct, then I will make the necessary modifications based on this understanding.

Reply:
if ENABLE_STDCXX_SYNC Is ON, of course, sorry, you're right :)

Still, you don't have to check for Android at all. You can mention in the comment that this situation is on some older versions of Android, but checking it isn't necessary.

So, if ENABLE_STDCXX_SYNC is ON, or if ENABLE_MONOTONIC_CLOCK is OFF, then monotonic clock is not requested.

If it is requested, need to check if both ENABLE_MONOTONIC_CLOCK_DEFAULT is ON and HAVE_PTHREAD_CONDATTR_SETCLOCK is ON, and if not, issue a fatal error. This check must be done after the declaration of the ENABLE_MONOTONIC_CLOCK option (because only there you can check its value as given by user or remaining as default).

Reply:
hmmm, Essentially, it should be consistent with what you said, but I’ve found that if my current platform is MacOS, then by default, ENABLE_MONOTONIC_CLOCK_DEFAULT is set to ON. However, when I check pthread_condattr_setclock, it indicates that the symbol is not found. But according to the definition in sync.h, even if ENABLE_STDCXX_SYNC is OFF, it does not use pthread on MacOS… At the root of the issue, it’s that we still define it when we are not using ENABLE_MONOTONIC_CLOCK…

There is detailed information in [3138#issuecomment-2721544858](https://github.com/Haivision/srt/pull/3138#issuecomment-2721544858) here.

Reply:
> hmmm, Essentially, it should be consistent with what you said, but I’ve found that if my current platform is MacOS, then by default, ENABLE_MONOTONIC_CLOCK_DEFAULT is set to ON. However, when I check pthread_condattr_setclock, it indicates that the symbol is not found. But according to the definition in sync.h, even if ENABLE_STDCXX_SYNC is OFF, it does not use pthread on MacOS… At the root of the issue, it’s that we still define it when we are not using ENABLE_MONOTONIC_CLOCK…
> 
> There is detailed information in [3138#issuecomment-2721544858](https://github.com/Haivision/srt/pull/3138#issuecomment-2721544858) here.

@ethouris But, I want to confirm if my understanding here is correct, because it concerns the necessity of the judgment for ‘Android’ in the final if statement